src/core/hle/service/http/http_c.cpp

```diff
@@ -147,12 +147,28 @@ static void ReceiveData(Service::Interface* self) {
 
     std::lock_guard<std::mutex> lock(map_it->second->mutex);
     const std::vector<u8>& data = map_it->second->response_data;
-    if (buf_size < data.size()) {
+
+    //TODO: Check if this should use context->downloaded_size instead of data.size()
+    if (buf_size > data.size()) {
         cmd_buff[1] = 0xD840A02B;
         return;
     }
 
-    std::copy(data.begin(), data.end(), buffer);
+    if (buf_size < data.size()) {
+        long data_size;
+
+        if (map_it->second->current_content_length + buf_size > map_it->second->content_length)
```

Comment:
Perhaps it'd be better to define a local variable for `map_it->second` so you don't have to repeat it so many times.